src/orderbook.ts

```diff
@@ -21,12 +21,20 @@ export class Offer {
   clone() {
     return new Offer(this.price.clone(), this.volume.clone());
   }
+
+  serialize(): object {
```

Comment:
As far as I understood [the spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#Description) `toJson` is supposed to contain the logic for serialization. I see that this implementation differs from the existing one in that it doesn't lose precision when serializing the Fraction.

Maybe we can change the `toJSON` logic in `Fraction` to retain the precision and then get rid of `serialize()`?

Reply:
The reason I didn't touch the existing `toJSON`s is that they seemed to intentionally not serialize the whole object. For example the orderbook's doesn't even include all of the class members. I was afraid I would break other code by changing this.
It looks like you think this isn't a problem so I will do the change.

Reply:
Just tried this: I encountered the problem that all of the tests use `stringify` as equality comparison. So me changing `toJSON` breaks the tests. Do you still want this even when it means changing the tests?

Reply:
I see the issue. I think it would still be cleaner to reuse the logic, add the remaining fields and update the tests (I realize this is a bit tedious).
It's kind of nice for the response of dex-price-estimator to contain what orderbook it's returning and what fee it is assuming.

Maybe we can reduce the fraction/volume before serializing it so that we get a canonical representation (e.g. 90/100 => 9/10). There is a `reduce` method on `Fraction`